source/postprocess/depth_average.cc

```diff
@@ -73,25 +73,126 @@ namespace aspect
+          for (unsigned int c=0; c<this->n_compositional_fields(); ++c)
+            variables.push_back(std::string("C_") + Utilities::int_to_string(c));
+          variables.push_back ("adiabatic_temperature");
+          variables.push_back ("velocity_magnitude");
+          variables.push_back ("sinking_velocity");
+          variables.push_back ("Vs");
+          variables.push_back ("Vp");
+          variables.push_back ("viscosity");
+          variables.push_back ("vertical_heat_flux");
+          n_statistics = variables.size();
+        }
+      else  //we have to parse the list in this order to match the output below
+        {
+          if ( std::find( output_variables.begin(), output_variables.end(), "temperature") != output_variables.end() )
+            {
+              n_statistics++;
+              variables.push_back("temperature");
+            }
+
+          if ( std::find( output_variables.begin(), output_variables.end(), "composition") != output_variables.end() )
+            {
+              n_statistics += this->n_compositional_fields();
+              for (unsigned int c=0; c<this->n_compositional_fields(); ++c)
+                variables.push_back(std::string("C_") + Utilities::int_to_string(c));
+            }
+
+          if ( std::find( output_variables.begin(), output_variables.end(), "adiabatic_temperature") != output_variables.end() )
+            {
+              n_statistics++;
+              variables.push_back("adiabatic_temperature");
+            }
+
+          if ( std::find( output_variables.begin(), output_variables.end(), "velocity_magnitude") != output_variables.end() )
+            {
+              n_statistics++;
+              variables.push_back("velocity_magnitude");
+            }
+
+          if ( std::find( output_variables.begin(), output_variables.end(), "sinking_velocity") != output_variables.end() )
+            {
+              n_statistics++;
+              variables.push_back("sinking_velocity");
+            }
+
+          if ( std::find( output_variables.begin(), output_variables.end(), "Vs") != output_variables.end() )
+            {
+              n_statistics++;
+              variables.push_back("Vs");
+            }
+
+          if ( std::find( output_variables.begin(), output_variables.end(), "Vp") != output_variables.end() )
+            {
+              n_statistics++;
+              variables.push_back("Vp");
+            }
+
+          if ( std::find( output_variables.begin(), output_variables.end(), "viscosity") != output_variables.end() )
+            {
+              n_statistics++;
+              variables.push_back("viscosity");
+            }
+
+          if ( std::find( output_variables.begin(), output_variables.end(), "vertical_heat_flux") != output_variables.end() )
+            {
+              n_statistics++;
+              variables.push_back("vertical_heat_flux");
+            }
+        }
+      Assert (variables.size() == n_statistics, ExcInternalError());
```

Comment:
why do you compute n_statistics separately above instead of using variables.size()?